src/rational/q/from.rs

```diff
@@ -449,6 +399,162 @@ mod tests_from_str {
 
 #[cfg(test)]
 mod test_from_int_int {
```

Comment:
Generally like extensive testing, but it looks a little too overklill, but I dont mind too much, lave it imo